app/javascript/multi-stamp-tubes/components/TubeArraySummary.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <table id="tube_scan_summary" :class="['plate-view', 'pool-colours']">
+    <caption>
+      {{
+        'Summary of tubes scanned into the rack'
+      }}
+    </caption>
+    <thead>
+      <tr>
+        <th class="first-col" />
+        <th id="header_human_barcode" class="headingcell">Human Barcode</th>
+        <th id="header_machine_barcode" class="headingcell">Machine Barcode</th>
+        <th id="header_replicates" class="headingcell">Replicates</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr v-for="(row, rowIndex) in tubeSummary" :key="row.barcode">
+        <th class="first-col">
+          {{ (rowIndex + 1).toString() + '.' }}
+        </th>
+        <td :id="`row_human_barcode_index_${rowIndex}`" class="summarycell">
+          {{ row.human_barcode }}
+        </td>
+        <td :id="`row_machine_barcode_index_${rowIndex}`" class="summarycell">
+          {{ row.machine_barcode }}
+        </td>
+        <td :id="`row_replicates_index_${rowIndex}`" class="replicate_cell">
+          {{ row.replicates }}
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+export default {
+  name: 'TubeArraySummary',
+  props: {
+    tubes: {
+      type: Array,
+      required: true,
+    },
+  },
+  computed: {
+    tubesDict() {
+      var summary_dict = {}
+      this.tubes.forEach(function (tube) {
+        var tube_machine_barcode = 'Empty'
+        var tube_human_barcode = 'Empty'
+
+        // skip empty locations
```

Comment:
If you're calling these 'locations'... does the `tubes` array actually represent the list of boxes on the parent component? Would it make more sense to call it `locations` or `wells`?

Reply:
We're passing in the computed 'tubes' value from the parent, I've got to call it tubes in this component so that it gets automatically updated when the parent computed value updates (on each change in any of the 96 tube scan input boxes). Vue JS reactivity.
The parent is including all 96 locations from the arraying page in the tubes array (i.e. it represents a tube rack really), and any that are empty are included in the tubes array with null values, so I am filtering these and displaying a count

Reply:
changed comments to make more sense hopefully